src/proc/sequence_id_filter.cpp

```diff
@@ -30,61 +26,53 @@ namespace librealsense
+        if (!frame.supports_frame_metadata(RS2_FRAME_METADATA_SEQUENCE_SIZE))
             return false;
-        if (!depth_frame.supports_frame_metadata(RS2_FRAME_METADATA_SUBPRESET_SEQUENCE_ID))
-            return false;
-        int depth_sequ_size = depth_frame.get_frame_metadata(RS2_FRAME_METADATA_SUBPRESET_SEQUENCE_SIZE);
-        if (depth_sequ_size == 0)
-            return false;
-
-        auto profile = frame.get_profile();
-        rs2_stream stream = profile.stream_type();
-        if (stream != RS2_STREAM_DEPTH)
+        if (!frame.supports_frame_metadata(RS2_FRAME_METADATA_SEQUENCE_ID))
             return false;
-
-        int index = profile.stream_index();
-        if (index != 0)
+        int sequ_size = frame.get_frame_metadata(RS2_FRAME_METADATA_SEQUENCE_SIZE);
+        if (sequ_size == 0)
             return false;
-
         return true;
     }
 
     rs2::frame sequence_id_filter::process_frame(const rs2::frame_source& source, const rs2::frame& f)
     {
         // steps:
-        // only for depth:
-        // 1. check hdr seq id in metadata -
-        //   if not as the option selected id, return last frame with the selected id
-        //   else return current frame
-
-        // 1. check hdr seq id in metadata
-        auto depth_frame = f.as<rs2::depth_frame>();
-        int seq_id = depth_frame.get_frame_metadata(RS2_FRAME_METADATA_SUBPRESET_SEQUENCE_ID);
+        // check hdr seq id in metadata -
+        // if not as the option selected id, return last frame with the selected id
+        // else return current frame
 
+        int seq_id = f.get_frame_metadata(RS2_FRAME_METADATA_SEQUENCE_ID);
+        auto profile = f.get_profile();
+        int frame_qualifier = calculate_frame_qualifier(seq_id, profile);
 
         if (is_selected_id(seq_id + 1))
         {
-            _last_frame[static_cast<int>(_selected_stream_id)] = f;
+            _last_frames[frame_qualifier] = f;
             return f;
         }
         else
         {
-            if (_last_frame[static_cast<int>(_selected_stream_id)])
-                return _last_frame[static_cast<int>(_selected_stream_id)];
+            int seq_id_selected = (seq_id == 0) ? 1 : 0;
+            int frame_qualifier_with_selected_id = calculate_frame_qualifier(seq_id_selected, profile);
+            if (_last_frames[frame_qualifier_with_selected_id])
+                return _last_frames[frame_qualifier_with_selected_id];
             return f;
         }
     }
 
-    bool sequence_id_filter::is_selected_id(int stream_index)
+    bool sequence_id_filter::is_selected_id(int stream_index) const
     {
         if (static_cast<int>(_selected_stream_id) != 0 &&
             stream_index != static_cast<int>(_selected_stream_id))
             return false;
         return true;
     }
+
+    int sequence_id_filter::calculate_frame_qualifier(int sequence_id, rs2::stream_profile profile) const
```

Comment:
Check and refactor with `frame->stream_profile->unique_id` - should be sufficient so you don't need the extra hash generator

Reply:
Done - pair<seq_id, unique_id> used as map's key